luci-app-firewall: add masq6 option for zones

1715173329 opened the pull request:
Allow configure Masquerading6 via LuCI interface.

jow- commented on applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js:
```diff
@@ -157,18 +157,31 @@ return view.extend({
 			var masq_src = uci.get('firewall', section_id, 'masq_src')
 			var masq_dest = uci.get('firewall', section_id, 'masq_dest')
-			if (masq_src || masq_dest)
+			if ((!family || family === 'ipv4') && (masq_src || masq_dest))
```
The check is overly strict, `family` may be `inet`, `inet6`, `ipv4`, `ipv6`, `4`, `6`. I suggest to do `family.indexOf('6') == -1`

1715173329 replied:
Got it, thanks!

1715173329 commented on applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js:
```diff
@@ -157,18 +157,31 @@ return view.extend({
 			return null;
 		};
 
+		o = s.taboption('general', form.Flag, 'masq6', _('IPv6 Masquerading'),
```
Sorry I don't even notice there's no masq6 support in fw3.. Should I also check firewall version for masq_src / masq_dest like this?

```js
if (!fw4) {
	o.depends('family', '');
	o.depends('family', 'ipv4');
	o.datatype = 'list(neg(or(uciname,hostname,ipmask4)))';
} else
	o.datatype = 'list(neg(or(uciname,hostname,ipmask)))';
```

jow- commented:
And one final remark; I am not sure if we should expose the `masq6` option this prominently. At the very least it should be hidden from the tabular zone overview (by removing `o.editable = true;`) and maybe it should be even moved into the advanced section.

IPv6 masquerading on a zone should be the exception (as last resort after PD and relay mode) rather than the norm.

1715173329 commented:
> And one final remark; I am not sure if we should expose the masq6 option this prominently. At the very least it should be hidden from the tabular zone overview (by removing o.editable = true;) and maybe it should be even moved into the advanced section.

Updated and moved it into the advanced section.

jow- commented:
Merged, thanks!
